src/main/java/org/broadinstitute/consent/http/db/mapper/DatasetReducer.java

```diff
@@ -40,12 +41,25 @@ public void accumulate(Map<Integer, Dataset> map, RowView rowView) {
         && hasColumn(rowView, "propertyvalue", String.class)) {
       String keyName = rowView.getColumn("key", String.class);
       String propVal = rowView.getColumn("propertyvalue", String.class);
+      DatasetPropertyType propType = DatasetPropertyType.String;
+      if (hasColumn(rowView, "property_type", String.class)) {
+          propType = DatasetPropertyType.parse(rowView.getColumn("property_type", String.class));
+      }
+
       if (Objects.nonNull(keyName) && Objects.nonNull(propVal)) {
-        DatasetProperty prop = new DatasetProperty();
-        prop.setDataSetId(dataset.getDataSetId());
-        prop.setPropertyValue(propVal);
-        prop.setPropertyName(keyName);
-        dataset.addProperty(prop);
+        try {
+          DatasetProperty prop = new DatasetProperty();
+          prop.setDataSetId(dataset.getDataSetId());
+          prop.setPropertyValue(propType.coerce(propVal));
+          prop.setPropertyName(keyName);
+          prop.setPropertyType(propType);
+          if (hasColumn(rowView, "schema_property", String.class)) {
+            prop.setSchemaProperty(rowView.getColumn("schema_property", String.class));
+          }
+          dataset.addProperty(prop);
+        } catch (Exception e) {
+          // do nothing.
+        }
```

Comment:
I wasn't able to comment on the actual line, but I think `name.setPropertyType(DatasetPropertyType.String))` should be added ~line 78 for when we manually add a name property

Reply:
great catch - just fixed!